cmake/stm32/devices.cmake

```diff
@@ -1116,14 +1116,68 @@ set(STM32_ALL_DEVICES
     WLE5JC
 )
 
-function(stm32_get_devices_by_family DEVICES)
+# Store a list of devices into a given STM_DEVICES list.
+# You can also specify multiple device families. Examples:
+# Get list of all devices for H7 family: stm32_get_devices_by_family(STM_DEVICES FAMILY H7)
+# Get list of all devices: stm32_get_devices_by_family(STM_DEVICES)
+function(stm32_get_devices_by_family STM_DEVICES)
+    # Specify keywords for argument parsing here
     set(ARG_OPTIONS "")
-    set(ARG_SINGLE FAMILY)
-    set(ARG_MULTIPLE "")
+    set(ARG_SINGLE "")
+    set(ARG_MULTIPLE FAMILY)
+
+    if(ARGC GREATER_EQUAL 1)
+        if(NOT ${ARGV0} STREQUAL "STM_DEVICES")
+            message(WARNING "Passed list ${ARGV0} invalid. Still setting STM_DEVICES variable")
+        endif()
```

Comment:
this is not what the function is doing. See https://markdewing.github.io/blog/posts/notes-on-cmake/

```
stm32_get_devices_by_family(retVal FAMILY L4)
message(STATUS "retVal is ${retVal}")
message(STATUS "STM_DEVICES is ${STM_DEVICES}")
```
outputs
```
[build] CMake Warning at C:/GIT/stm32-cmake/cmake/stm32/devices.cmake:1131 (message):
[build]   Passed list retVal invalid.  Still setting STM_DEVICES variable
[build] Call Stack (most recent call first):
[build]   CMakeLists.txt:8 (stm32_get_devices_by_family)
[build] 
[build] 
[build] -- retVal is L412C8;L412CB;L412K8;L412KB;L412R8;L412RB;L412T8;L412TB;L422CB;L422KB;L422RB;L422TB;L431CB;L431CC;L431KB;L431KC;L431RB;L431RC;L431VC;L432KB;L432KC;L433CB;L433CC;L433RB;L433RC;L433VC;L442KC;L443CC;L443RC;L443VC;L451CC;L451CE;L451RC;L451RE;L451VC;L451VE;L452CC;L452CE;L452RC;L452RE;L452VC;L452VE;L462CE;L462RE;L462VE;L471QE;L471QG;L471RE;L471RG;L471VE;L471VG;L471ZE;L471ZG;L475RC;L475RE;L475RG;L475VC;L475VE;L475VG;L476JE;L476JG;L476ME;L476MG;L476QE;L476QG;L476RC;L476RE;L476RG;L476VC;L476VE;L476VG;L476ZE;L476ZG;L486JG;L486QG;L486RG;L486VG;L486ZG;L496AE;L496AG;L496QE;L496QG;L496RE;L496RG;L496VE;L496VG;L496ZE;L496ZG;L4A6AG;L4A6QG;L4A6RG;L4A6VG;L4A6ZG;L4P5AE;L4P5AG;L4P5CE;L4P5CG;L4P5QE;L4P5QG;L4P5RE;L4P5RG;L4P5VE;L4P5VG;L4P5ZE;L4P5ZG;L4Q5AG;L4Q5CG;L4Q5QG;L4Q5RG;L4Q5VG;L4Q5ZG;L4R5AG;L4R5AI;L4R5QG;L4R5QI;L4R5VG;L4R5VI;L4R5ZG;L4R5ZI;L4R7AI;L4R7VI;L4R7ZI;L4R9AG;L4R9AI;L4R9VG;L4R9VI;L4R9ZG;L4R9ZI;L4S5AI;L4S5QI;L4S5VI;L4S5ZI;L4S7AI;L4S7VI;L4S7ZI;L4S9AI;L4S9VI;L4S9ZI
[build] -- STM_DEVICES is 
```